spock-core/src/main/java/org/spockframework/compiler/IRewriteResources.java

```diff
@@ -17,14 +17,27 @@
 package org.spockframework.compiler;
 
 import org.codehaus.groovy.ast.ASTNode;
-import org.spockframework.compiler.condition.IConditionErrorRecorders;
+import org.codehaus.groovy.ast.expr.Expression;
+import org.codehaus.groovy.ast.expr.MethodCallExpression;
+import org.codehaus.groovy.ast.expr.VariableExpression;
 
-import java.util.List;
+import org.spockframework.compiler.condition.IConditionErrorRecorders;
+import org.spockframework.compiler.model.Block;
+import org.spockframework.compiler.model.Method;
 
 /**
+ *
  * @author Peter Niederwieser
  */
 public interface IRewriteResources {
+  Method getCurrentMethod();
+
+  Block getCurrentBlock();
+
+  VariableExpression captureOldValue(Expression oldValue);
```

Comment:
These don't apply outside of a spec, though.

Reply:
Shouldn't we at least use a more useful error message in that case, i.e., that it is only supported by `SpecRewriter`/ within a Specification?

Reply:
Ok, I changed it to `throw new UnsupportedOperationException("This should only be called when processing a then-block");` for the old-case.

For the mock invocation matcher I didn't because I lied. :-)
That I did not already throw an UOE there is, because it does get called, but then the result is ignored in case of a `VerifyBlock`.
And the order cannot be swapped, because the result is used for an up-front check whether it is an interaction at all, before then complaining that you must not have one in expect or verify.